src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java

```diff
@@ -130,6 +130,11 @@ public ParticipationResource(ParticipationService participationService, Programm
             }
         }
 
+        // only instructors should be allowed to start an exam exercise using this resource e.g. to create the assignment repository
+        if (exercise.hasExerciseGroup() && !authCheckService.isAtLeastInstructorInCourse(course, user)) {
```

Comment:
This is wrong. We have a backup to start programming exercises during an exam in case something went wrong in the initial setup, see `exam-participation.component.ts:398`

Reply:
Missed that. I removed the check and the new corresponding tests for it